pkg/cmd/experimental/router/router.go

```diff
@@ -301,3 +316,19 @@ func RunCmdRouter(f *clientcmd.Factory, cmd *cobra.Command, out io.Writer, cfg *
 	fmt.Fprintf(out, "Router %q service exists\n", name)
 	return nil
 }
+
+// getServiceNameFromObjects operates on the assumption that there will only be a single service
+// in the list of objects that it can use to give back a namespace and name to be used in the
+// router's environment if it needs to reference its peers
+func getServiceNameFromObjects(objects []runtime.Object) (string, error) {
```

Comment:
Why is this necessary?  The service name should be the same as the deployment config name always.